src/main/java/org/dependencytrack/util/VulnerabilityPolicyUtil.java

```diff
@@ -0,0 +1,169 @@
+                        etag,
+                        ConfigPropertyConstants.VULNERABILITY_POLICY_FILE_LAST_MODIFIED_HASH.getPropertyType(),
+                        ConfigPropertyConstants.VULNERABILITY_POLICY_FILE_LAST_MODIFIED_HASH.getDescription());
+                return true;
+            } else if (lastModifiedHash.getPropertyValue() == null ||
+                    !(etag.equals(lastModifiedHash.getPropertyValue()))) {
+                //lastmodifiedhash value is either null or different from incoming value
+                lastModifiedHash.setPropertyValue(etag);
+                return true;
+            }
+            //the lastmodified hash is the same as the incoming value
+            return false;
+        }
+    }
+
+    public static void saveParsedVulnerabilities(QueryManager queryManager, List<VulnerabilityPolicy> createVulnerabilityPolicyList,
+                                           List<VulnerabilityPolicy> updateVulnerabilityPolicyList,
+                                           List<String> policyNames) {
+        List<String> receivedPolicyNames = new ArrayList<>();
+        for (var vulnerabilityPolicy : createVulnerabilityPolicyList) {
+            receivedPolicyNames.add(vulnerabilityPolicy.getName());
+        }
+        for (var vulnerabilityPolicy : updateVulnerabilityPolicyList) {
+            receivedPolicyNames.add(vulnerabilityPolicy.getName());
+        }
+        List<String> vulnerabilityPoliciesToBeDeleted = policyNames.stream()
+                .filter(policyName -> !receivedPolicyNames.contains(policyName)).toList();
+        queryManager.runInTransaction(() -> {
+            Connection connection = null;
+            try {
+                connection = (Connection) queryManager.getPersistenceManager().getDataStoreConnection();
+                for (var vulnerabilityPolicy : createVulnerabilityPolicyList) {
+                    queryManager.createVulnerabilityPolicy(vulnerabilityPolicy, connection);
+                }
+                for (var vulnerabilityPolicy : updateVulnerabilityPolicyList) {
+                    queryManager.updateVulnerablePolicyByName(vulnerabilityPolicy, connection);
+                }
+                for (var name : vulnerabilityPoliciesToBeDeleted) {
+                    queryManager.deleteVulnerabilityPolicyByName(name, connection);
+                }
+            } finally {
+                DbUtil.close(connection);
+            }
+
+        });
+    }
+
+    private static JsonSchema enrichYamlOjMapperAndGetSchema() throws IOException {
+        final String jsonSchemaContent = resourceToString("/schema/vulnerability-policy-v1.schema.json", StandardCharsets.UTF_8);
+        JsonSchemaFactory factory = JsonSchemaFactory.builder(JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V202012)).objectMapper(MAPPER).build();
+        return factory.getSchema(jsonSchemaContent);
+    }
```

Comment:
This is a good example of something that could be made a `private static final` field instead. Construction is kind of expensive (as it involves I/O and file parsing of the schema), and it's used multiple times (once per policy file).

Re-creating this for each policy file is unnecessary overhead and is best avoided.